File: leetCode/easy/longestCommonPrefix.js
function longestCommonPrefix (strs){
    commonPrefix = ""
    for (let i = 0; i < strs[0].length; i++) {
        for (let j = 0; j < strs.length; j++) {
            if (strs[i][j] === strs[0][i]){
                commonPrefix += strs[0][i]
            }
        }
        
    }

    return commonPrefix

}

console.log(longestCommonPrefix(["flower","flow","flight"]))

File: leetCode/easy/longestCommonPrefix.py
def longestCommonPrefix(strs):

    
    if strs == [""]:
        return ""
    
    if len(strs) == 1:
        return strs

    compare = min(strs, key=len)
    res = ""

    index = 0
    while index < len(compare):
        for word in strs:
            if word[index] != compare[index]:
                return res
        res += compare[index]
        index += 1



print(longestCommonPrefix(["a"]))